aten/src/ATen/native/mps/operations/Copy.mm

```diff
@@ -318,6 +318,41 @@ void copy_blit_mps(void* dst, const void* src, size_t size) {
   return dst_;
 }
 
+// tries to copy scalar buffers if any of them are allocated on MPSAllocator
+// with Shared-storage mode on Unified devices (supports CPU <-> MPS copies only).
+static bool try_copy_scalars_mps(at::Tensor& dst, const at::Tensor& src, bool non_blocking)
+{
+  const auto& allocator = *getIMPSAllocator();
+  if (!allocator.isSharedStorageSupported()) {
+    return false;
+  }
+  void* src_ptr = src.storage().data();
+  void* dst_ptr = dst.storage().data();
+  uint32_t retainCount = 0;
+
+  if (src.device().type() == at::kMPS) {
+    std::tie(src_ptr, retainCount) = allocator.getSharedBufferPtr(src_ptr);
+  }
+  if (dst.device().type() == at::kMPS) {
+    std::tie(dst_ptr, retainCount) = allocator.getSharedBufferPtr(dst_ptr);
```

Comment:
This is not making sense to me. Above on line 334 you have src retain count and now this is getting over written. Shouldn't you maintain src and dst retain counts separately ? Also the sync is needed only for src_retain_count > 1. Are there cases where you need to worry about dst_retain_count as well ?

Reply:
As mentioned in functions comments, this function is supported only for MPS to CPU and CPU to MPS copies. So one of them will be MPS at each time.

Reply:
I update the PR, in case in future we want to use it for MPS to MPS.